include/hal/rtc.h

```diff
@@ -3,32 +3,90 @@
 
 
 #include <stdint.h>
-#include "utils/queue.h"
 
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#define __HAL_DISABLE_INTERRUPTS(x)                     \
+    do {                                                \
+        x = __get_PRIMASK();                            \
+        __disable_irq();                                \
+    } while(0);
```

Comment:
I do not want this code here. The rtc.h file is the common API for all devices defied in the RTC0 folder. These things are device specific.

Reply:
So, Would you mind if I move those Macros to freertos.c source file ?